src/api/sirius_api.cpp

```diff
@@ -5996,6 +5996,24 @@ void sirius_linear_solver(void* const* handler__, double const* vkq__, int const
             auto Hphi_wf = sirius::wave_function_factory<double>(sctx, kp, wf::num_bands(sctx.num_bands()), wf::num_mag_dims(0), false);
             auto Sphi_wf = sirius::wave_function_factory<double>(sctx, kp, wf::num_bands(sctx.num_bands()), wf::num_mag_dims(0), false);
 
+            sddk::memory_t mem = sctx.processing_unit_memory_t(); // "auto mem" should do as well
+
+            // TODO: allocate all wave-functions on device (if processing_unit == sdk::memory_t::GPU)
+//          if (sctx.processing_unit() == sddk::device_t::GPU){
+            { // START scope of memory guards
+
```

Comment:
It is better to use std::vector<memory_guard> and push to it

Reply:
done